fda/grid.py

```diff
@@ -543,91 +547,151 @@ def concatenate(self, other):
                          self.dataset_label,
                          self.axes_labels)
 
-    def _set_labels(self, ax):
+    def _set_labels(self, fig):
+        # TODO: check in init the length of the labels
         """Set labels if any.
 
         Args:
-            ax (axes object): axes object that implements set_title,
-                set_xlable and set_ylabel or title, xlabel and ylabel.
+            fig (figure object): figure object containing the axes that implement set_xlabel and set_ylabel,
+                                 and set_zlabel in case of a 3d projection.
 
         """
         if self.dataset_label is not None:
-            try:
-                ax.set_title(self.dataset_label)
-            except AttributeError:
-                try:
-                    ax.title(self.dataset_label)
-                except AttributeError:
-                    pass
+            fig.suptitle(self.dataset_label, fontsize="x-large")
```

Comment:
Maybe it would be useful to allow changing plot parameters, such as the fontsize.